"libpas test suit is broken due to recent profile and track allocation mode change"

nishajain61 opened the pull request:
#### 51f42fcfcb404e74af4070bf068805155b889178
<pre>
&quot;libpas test suit is broken due to recent profile and track allocation mode change&quot;
<a href="https://bugs.webkit.org/show_bug.cgi?id=274554">https://bugs.webkit.org/show_bug.cgi?id=274554</a>
<a href="https://rdar.apple.com/128497967">rdar://128497967</a>

Reviewed by Yusuke Suzuki.

libpas test build fails due to recent change (122419407).
Fixed the files to add allocation mode..

* Source/bmalloc/libpas/src/chaos/Chaos.cpp:
(std::threadMain):
* Source/bmalloc/libpas/src/libpas/hotbit_heap.c:
(hotbit_try_allocate):
(hotbit_try_allocate_with_alignment):
(hotbit_try_reallocate):
* Source/bmalloc/libpas/src/libpas/hotbit_heap.h:
* Source/bmalloc/libpas/src/libpas/hotbit_heap_inlines.h:
(hotbit_try_allocate_inline):
(hotbit_try_allocate_with_alignment_inline):
(hotbit_try_reallocate_inline):
* Source/bmalloc/libpas/src/libpas/iso_heap.c:
(iso_try_allocate_common_primitive):
(iso_try_allocate_common_primitive_with_alignment):
(iso_try_allocate_common_primitive_zeroed):
(iso_allocate_common_primitive):
(iso_allocate_common_primitive_with_alignment):
(iso_allocate_common_primitive_zeroed):
(iso_try_reallocate_common_primitive):
(iso_reallocate_common_primitive):
(iso_try_allocate_dynamic_primitive):
(iso_try_allocate_dynamic_primitive_with_alignment):
(iso_try_allocate_dynamic_primitive_zeroed):
(iso_try_reallocate_dynamic_primitive):
(iso_try_allocate):
(iso_allocate):
(iso_try_allocate_array_by_count):
(iso_allocate_array_by_count):
(iso_try_allocate_array_by_count_zeroed):
(iso_allocate_array_by_count_zeroed):
(iso_try_reallocate_array_by_count):
(iso_reallocate_array_by_count):
(iso_try_allocate_primitive):
(iso_allocate_primitive):
(iso_try_allocate_primitive_zeroed):
(iso_allocate_primitive_zeroed):
(iso_try_allocate_primitive_with_alignment):
(iso_allocate_primitive_with_alignment):
(iso_try_reallocate_primitive):
(iso_reallocate_primitive):
(iso_try_allocate_for_flex):
* Source/bmalloc/libpas/src/libpas/iso_heap.h:
* Source/bmalloc/libpas/src/libpas/iso_heap_inlines.h:
(iso_try_allocate_common_primitive_inline):
(iso_try_allocate_common_primitive_with_alignment_inline):
(iso_try_allocate_common_primitive_zeroed_inline):
(iso_allocate_common_primitive_inline):
(iso_allocate_common_primitive_with_alignment_inline):
(iso_allocate_common_primitive_zeroed_inline):
(iso_try_reallocate_common_primitive_inline):
(iso_reallocate_common_primitive_inline):
(iso_try_allocate_inline):
(iso_allocate_inline):
(iso_try_allocate_array_by_count_inline):
(iso_allocate_array_by_count_inline):
(iso_try_allocate_array_by_count_zeroed_inline):
(iso_allocate_array_by_count_zeroed_inline):
(iso_try_reallocate_array_by_count_inline):
(iso_reallocate_array_by_count_inline):
(iso_try_allocate_primitive_inline):
(iso_allocate_primitive_inline):
(iso_try_allocate_primitive_zeroed_inline):
(iso_allocate_primitive_zeroed_inline):
(iso_try_allocate_primitive_with_alignment_inline):
(iso_allocate_primitive_with_alignment_inline):
(iso_try_reallocate_primitive_inline):
(iso_reallocate_primitive_inline):
(iso_try_allocate_for_flex_inline):
* Source/bmalloc/libpas/src/libpas/iso_test_heap.c:
(iso_test_allocate_common_primitive):
(iso_test_allocate):
(iso_test_allocate_array_by_count):
* Source/bmalloc/libpas/src/libpas/iso_test_heap.h:
* Source/bmalloc/libpas/src/libpas/jit_heap.c:
(jit_heap_try_allocate):
* Source/bmalloc/libpas/src/libpas/jit_heap.h:
* Source/bmalloc/libpas/src/libpas/minalign32_heap.c:
(minalign32_allocate_common_primitive):
(minalign32_allocate):
(minalign32_allocate_array_by_count):
* Source/bmalloc/libpas/src/libpas/minalign32_heap.h:
* Source/bmalloc/libpas/src/libpas/pagesize64k_heap.c:
(pagesize64k_allocate_common_primitive):
(pagesize64k_allocate):
(pagesize64k_allocate_array_by_count):
* Source/bmalloc/libpas/src/libpas/pagesize64k_heap.h:
* Source/bmalloc/libpas/src/libpas/pas_allocation_mode.h:
* Source/bmalloc/libpas/src/libpas/thingy_heap.c:
(thingy_try_allocate_primitive):
(thingy_try_allocate_primitive_zeroed):
(thingy_try_allocate_primitive_with_alignment):
(thingy_try_reallocate_primitive):
(thingy_try_allocate):
(thingy_try_allocate_zeroed):
(thingy_try_allocate_array):
(thingy_try_allocate_zeroed_array):
(thingy_try_reallocate_array):
(thingy_utility_heap_allocate):
* Source/bmalloc/libpas/src/libpas/thingy_heap.h:
* Source/bmalloc/libpas/src/libpas/thingy_heap_prefix.h:
* Source/bmalloc/libpas/src/mbmalloc/mbmalloc_bmalloc.c:
(mbmalloc):
(mbmemalign):
(mbrealloc):
* Source/bmalloc/libpas/src/mbmalloc/mbmalloc_hotbit.c:
(mbmalloc):
(mbmemalign):
(mbrealloc):
* Source/bmalloc/libpas/src/mbmalloc/mbmalloc_iso_common_primitive.c:
(mbmalloc):
(mbmemalign):
(mbrealloc):
* Source/bmalloc/libpas/src/test/BitfitTests.cpp:
(std::testAllocateAlignedSmallerThanSizeClassAndSmallerThanLargestAvailable):
* Source/bmalloc/libpas/src/test/BmallocTests.cpp:
(std::testBmallocAllocate):
(std::testBmallocDeallocate):
* Source/bmalloc/libpas/src/test/EnumerationTests.cpp:
(std::testBasicEnumeration):
(std::testPGMEnumerationBasic):
(std::testPGMEnumerationAddAndFree):
* Source/bmalloc/libpas/src/test/ExpendableMemoryTests.cpp:
(std::testPayloadImpl):
(std::testRage):
(std::testRematerializeAfterSearchOfDecommitted):
(std::testBasicSizeClass):
* Source/bmalloc/libpas/src/test/HeapRefAllocatorIndexTests.cpp:
* Source/bmalloc/libpas/src/test/IsoDynamicPrimitiveHeapTests.cpp:
(std::allocate42):
(std::allocate42WithAlignment):
(std::allocate42Zeroed):
(std::reallocate42):
* Source/bmalloc/libpas/src/test/IsoHeapChaosTests.cpp:
(std::gigacageAllocate):
(std::testAllocationChaos):
* Source/bmalloc/libpas/src/test/IsoHeapPageSharingTests.cpp:
(std::testTakePages):
(std::testTakePagesFromCorrectHeap):
(std::testLargeHeapTakesPagesFromCorrectSmallHeap):
(std::testLargeHeapTakesPagesFromCorrectSmallHeapAllocateAfterFree):
(std::testLargeHeapTakesPagesFromCorrectSmallHeapWithFancyOrder):
(std::testLargeHeapTakesPagesFromCorrectLargeHeap):
(std::testLargeHeapTakesPagesFromCorrectLargeHeapAllocateAfterFreeOnSmallHeap):
(std::testLargeHeapTakesPagesFromCorrectLargeHeapAllocateAfterFreeOnAnotherLargeHeap):
(std::testLargeHeapTakesPagesFromCorrectLargeHeapWithFancyOrder):
(std::testSmallHeapTakesPagesFromCorrectLargeHeap):
(std::testSmallHeapTakesPagesFromCorrectLargeHeapWithFancyOrder):
(std::testSmallHeapTakesPagesFromCorrectLargeHeapAllocateAfterFreeOnSmallHeap):
(std::testSmallHeapTakesPagesFromCorrectLargeHeapAllocateAfterFreeOnAnotherLargeHeap):
(std::allocateThingiesImpl):
(std::testScavengerEventuallyReturnsMemory):
(std::testScavengerEventuallyReturnsMemoryEvenWithoutManualShrink):
* Source/bmalloc/libpas/src/test/IsoHeapPartialAndBaselineTests.cpp:
(std::testSimplePartialAllocations):
(std::testFreeAroundPrimordialStop):
(std::testFreeInterleavedAroundPrimordialStop):
(std::testMultiplePartialsFromDifferentHeapsPerShared):
(std::testMultiplePartialsFromDifferentThreadsPerShared):
(std::testTwoBaselinesEvictions):
* Source/bmalloc/libpas/src/test/IsoHeapReservedMemoryTests.cpp:
(std::testSizeProgression):
* Source/bmalloc/libpas/src/test/JITHeapTests.cpp:
(std::testAllocateShrinkAndAllocate):
(std::testAllocationSize):
* Source/bmalloc/libpas/src/test/LotsOfHeapsAndThreads.cpp:
(std::testLotsOfHeapsAndThreads):
* Source/bmalloc/libpas/src/test/MemalignTests.cpp:
* Source/bmalloc/libpas/src/test/PGMTests.cpp:
(std::testPGMSingleAlloc):
(std::testPGMMultipleAlloc):
(std::testPGMRealloc):
(std::testPGMErrors):
* Source/bmalloc/libpas/src/test/RaceTests.cpp:
(std::testLocalAllocatorStopRace):
(std::testLocalAllocatorStopRaceAgainstScavenge):
* Source/bmalloc/libpas/src/test/TLCDecommitTests.cpp:
(std::testTLCDecommit):
(std::testChaosThenDecommit):
(std::testAllocateFromStoppedBaselineImpl):
* Source/bmalloc/libpas/src/test/TSDTests.cpp:
(std::destructor):
(std::testTSD):
* Source/bmalloc/libpas/src/test/ThingyAndUtilityHeapAllocationTests.cpp:
(std::testFreeListRefillSpans):
(std::testInternalScavenge):
(std::testInternalScavengeFromCorrectDirectory):
(std::testSpuriousEligibility):
(std::testBasicSizeClassNotSet):
(std::testSmallDoubleFree):
(std::testSmallFreeInner):
(std::testSmallFreeNextWithShrink):
(std::testSmallFreeNextWithoutShrink):
(std::testSmallFreeNextBeforeShrink):
(std::testAllocationChaos):
(std::testCombinedAllocationChaos):
(std::testLargeDoubleFree):
(std::testLargeOffsetFree):
(std::testReallocatePrimitive):
(std::testReallocateArray):
* Source/bmalloc/libpas/src/test/ViewCacheTests.cpp:
(std::testDisableViewCacheUsingBoundForNoViewCache):
(std::testEnableViewCacheAtSomeBoundForNoViewCache):

Canonical link: <a href="https://commits.webkit.org/279280@main">https://commits.webkit.org/279280@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/24b204039150544fbd8adf847705142ca9852cf7

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/52869 "359 style errors") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/32206 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/5356 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/56148 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/3592 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/39050 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/3317 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/42912 "Passed tests") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/2322 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/54967 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/29998 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/45641 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/24028 "Passed tests") | 
| | [❌ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/27061 "4 new passes 4 flakes") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/2959 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/1751 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/46223 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/48916 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/3113 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/57741 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/52382 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/28010 "Built successfully") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/3087 "Found 1 new test failure: imported/w3c/web-platform-tests/html/browsers/the-window-object/open-close/open-features-tokenization-noreferrer.html (failure)") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/50301 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/29230 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/45844 "Passed tests") | [❌ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/49589 "Found 2 new API test failures: /WebKitGTK/TestWebKitWebContext:/webkit/WebKitWebContext/memory-pressure, /WebKitGTK/TestWebKitAccessibility:/webkit/WebKitAccessibility/text/state-changed (failure)") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/11569 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/30149 "Built successfully") | | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/64686 "Built successfully") | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/28984 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/12278 "Passed tests") | 
<!--EWS-Status-Bubble-End-->

ddegazio commented on Source/bmalloc/libpas/src/libpas/jit_heap.c:
```diff
@@ -72,13 +72,13 @@ PAS_CREATE_TRY_ALLOCATE_INTRINSIC(
     pas_intrinsic_heap_is_not_designated);
 
-void* jit_heap_try_allocate(size_t size)
+void* jit_heap_try_allocate(size_t size, pas_allocation_mode allocation_mode)
```
We want to treat all allocations in the JIT heap as `compact`, so we don't need to pass in an `allocation_mode` here. Let's just directly call `jit_try_allocate_common_primitive_impl` with `pas_compact_allocation_mode`.

hyjorc1 replied:
What's the `compact` ptr here? Is that using 32bit to represent 36bit address for 16byte aligned classes. Why all allocations in the JIT heap are using `CompactPtr`? Is it used when allocating 16byte aligned objects?

ddegazio commented on Source/bmalloc/libpas/src/libpas/jit_heap.h:
```diff
@@ -44,7 +45,7 @@ PAS_API extern pas_allocator_counts jit_allocator_counts;
 PAS_API void jit_heap_add_fresh_memory(pas_range range);
 
-PAS_API void* jit_heap_try_allocate(size_t size);
+PAS_API void* jit_heap_try_allocate(size_t size, pas_allocation_mode allocation_mode);
```
Same as above; let's avoid having an `allocation_mode` parameter for the JIT heap.

ddegazio commented on Source/bmalloc/libpas/src/mbmalloc/mbmalloc_bmalloc.c:
```diff
@@ -35,17 +35,17 @@
 void* mbmalloc(size_t size)
 {
-    return bmalloc_try_allocate(size);
+    return bmalloc_try_allocate(size, 1);
```
Let's make these `pas_non_compact_allocation_mode` to be more explicit.

ddegazio commented on Source/bmalloc/libpas/src/mbmalloc/mbmalloc_hotbit.c:
```diff
@@ -35,17 +35,17 @@
 void* mbmalloc(size_t size)
 {
-    return hotbit_try_allocate(size);
+    return hotbit_try_allocate(size, 1);
```
Ditto.

ddegazio commented on Source/bmalloc/libpas/src/mbmalloc/mbmalloc_iso_common_primitive.c:
```diff
@@ -69,23 +69,23 @@ void* mbmalloc(size_t size)
         printf("malloc(%zu)\n", size);
     install_verifier_if_necessary();
-    return iso_try_allocate_common_primitive(size);
+    return iso_try_allocate_common_primitive(size, 1);
```
Ditto.

ddegazio commented on Source/bmalloc/libpas/src/test/JITHeapTests.cpp:
```diff
@@ -44,13 +44,13 @@ void testAllocateShrinkAndAllocate(unsigned initialObjectSize,
 {
     for (unsigned i = numInitialObjects; i--;) {
-        void* ptr = jit_heap_try_allocate(initialObjectSize);
+        void* ptr = jit_heap_try_allocate(initialObjectSize, pas_compact_allocation_mode);
```
If we remove the `allocation_mode` parameter from `jit_heap_try_allocate`, we won't need to pass `pas_compact_allocation_mode`.

ddegazio commented on Source/bmalloc/libpas/src/test/JITHeapTests.cpp:
```diff
@@ -44,13 +44,13 @@ void testAllocateShrinkAndAllocate(unsigned initialObjectSize,
     PAS_ASSERT(sizeToShrinkTo <= initialSizeOfObjectToShrink);
 
-    void* ptr = jit_heap_try_allocate(initialSizeOfObjectToShrink);
+    void* ptr = jit_heap_try_allocate(initialSizeOfObjectToShrink, pas_compact_allocation_mode);
```
Ditto.

ddegazio commented on Source/bmalloc/libpas/src/test/JITHeapTests.cpp:
```diff
@@ -59,7 +59,7 @@ void testAllocateShrinkAndAllocate(unsigned initialObjectSize,
         expectedAlignmentOfFinalObject);
 
-    void* newPtr = jit_heap_try_allocate(finalObjectSize);
+    void* newPtr = jit_heap_try_allocate(finalObjectSize, pas_compact_allocation_mode);
```
Ditto.

ddegazio commented on Source/bmalloc/libpas/src/test/JITHeapTests.cpp:
```diff
@@ -71,7 +71,7 @@ void testAllocateShrinkAndAllocate(unsigned initialObjectSize,
 void testAllocationSize(size_t requestedSize, size_t actualSize)
 {
-    CHECK_EQUAL(jit_heap_get_size(jit_heap_try_allocate(requestedSize)), actualSize);
+    CHECK_EQUAL(jit_heap_get_size(jit_heap_try_allocate(requestedSize, pas_compact_allocation_mode)), actualSize);
```
Ditto.

ddegazio commented on Source/bmalloc/libpas/src/test/LotsOfHeapsAndThreads.cpp:
```diff
@@ -48,7 +48,7 @@ void testLotsOfHeapsAndThreads(unsigned numHeaps, unsigned numThreads, unsigned
             for (unsigned j = count; j--;) {
                 for (unsigned k = numHeaps; k--;) {
-                    void* ptr = bmalloc_iso_allocate(heaps + k);
+                    void* ptr = bmalloc_iso_allocate((heaps + k), pas_non_compact_allocation_mode);
```
We can just leave `(heaps + k)` as `heaps + k`.

ddegazio commented on Source/bmalloc/libpas/src/test/TLCDecommitTests.cpp:
```diff
@@ -72,7 +72,7 @@ void testTLCDecommit(unsigned numHeaps,
     vector<void*> objects;
     for (size_t index = 0; index < numHeaps; ++index) {
-        void* ptr = bmalloc_iso_allocate(heaps + index);
+        void* ptr = bmalloc_iso_allocate((heaps + index),pas_non_compact_allocation_mode);
```
Unnecessary parentheses around `heaps + index`.

ddegazio commented on Source/bmalloc/libpas/src/test/TLCDecommitTests.cpp:
```diff
@@ -172,7 +172,7 @@ void testTLCDecommit(unsigned numHeaps,
 
     for (size_t index = 0; index < numHeaps; ++index) {
-        void* ptr = bmalloc_iso_allocate(heaps + index);
+        void* ptr = bmalloc_iso_allocate((heaps + index), pas_non_compact_allocation_mode);
```
Ditto.

ddegazio commented on Source/bmalloc/libpas/src/test/TLCDecommitTests.cpp:
```diff
@@ -269,7 +269,7 @@ vector<void*> prepareToTestDLCDecommitThenStuff(unsigned numHeaps)
                  512, BMALLOC_HEAP_CONFIG.small_segregated_config, pas_segregated_page_exclusive_role);
              objectIndex--;) {
-            void* ptr = bmalloc_iso_allocate(heaps + index);
+            void* ptr = bmalloc_iso_allocate((heaps + index), pas_non_compact_allocation_mode);
```
Ditto.

ddegazio reviewed (approved):
r=me with changes and if it passes EWS (looks like there's some build failures).

Constellation reviewed (approved):
r=me

nishajain61 committed:
"libpas test suit is broken due to recent profile and track allocation mode change"
https://bugs.webkit.org/show_bug.cgi?id=274554
rdar://128497967

Reviewed by Yusuke Suzuki.

libpas test build fails due to recent change (122419407).
Fixed the files to add allocation mode..

* Source/bmalloc/libpas/src/chaos/Chaos.cpp:
(std::threadMain):
* Source/bmalloc/libpas/src/libpas/hotbit_heap.c:
(hotbit_try_allocate):
(hotbit_try_allocate_with_alignment):
(hotbit_try_reallocate):
* Source/bmalloc/libpas/src/libpas/hotbit_heap.h:
* Source/bmalloc/libpas/src/libpas/hotbit_heap_inlines.h:
(hotbit_try_allocate_inline):
(hotbit_try_allocate_with_alignment_inline):
(hotbit_try_reallocate_inline):
* Source/bmalloc/libpas/src/libpas/iso_heap.c:
(iso_try_allocate_common_primitive):
(iso_try_allocate_common_primitive_with_alignment):
(iso_try_allocate_common_primitive_zeroed):
(iso_allocate_common_primitive):
(iso_allocate_common_primitive_with_alignment):
(iso_allocate_common_primitive_zeroed):
(iso_try_reallocate_common_primitive):
(iso_reallocate_common_primitive):
(iso_try_allocate_dynamic_primitive):
(iso_try_allocate_dynamic_primitive_with_alignment):
(iso_try_allocate_dynamic_primitive_zeroed):
(iso_try_reallocate_dynamic_primitive):
(iso_try_allocate):
(iso_allocate):
(iso_try_allocate_array_by_count):
(iso_allocate_array_by_count):
(iso_try_allocate_array_by_count_zeroed):
(iso_allocate_array_by_count_zeroed):
(iso_try_reallocate_array_by_count):
(iso_reallocate_array_by_count):
(iso_try_allocate_primitive):
(iso_allocate_primitive):
(iso_try_allocate_primitive_zeroed):
(iso_allocate_primitive_zeroed):
(iso_try_allocate_primitive_with_alignment):
(iso_allocate_primitive_with_alignment):
(iso_try_reallocate_primitive):
(iso_reallocate_primitive):
(iso_try_allocate_for_flex):
* Source/bmalloc/libpas/src/libpas/iso_heap.h:
* Source/bmalloc/libpas/src/libpas/iso_heap_inlines.h:
(iso_try_allocate_common_primitive_inline):
(iso_try_allocate_common_primitive_with_alignment_inline):
(iso_try_allocate_common_primitive_zeroed_inline):
(iso_allocate_common_primitive_inline):
(iso_allocate_common_primitive_with_alignment_inline):
(iso_allocate_common_primitive_zeroed_inline):
(iso_try_reallocate_common_primitive_inline):
(iso_reallocate_common_primitive_inline):
(iso_try_allocate_inline):
(iso_allocate_inline):
(iso_try_allocate_array_by_count_inline):
(iso_allocate_array_by_count_inline):
(iso_try_allocate_array_by_count_zeroed_inline):
(iso_allocate_array_by_count_zeroed_inline):
(iso_try_reallocate_array_by_count_inline):
(iso_reallocate_array_by_count_inline):
(iso_try_allocate_primitive_inline):
(iso_allocate_primitive_inline):
(iso_try_allocate_primitive_zeroed_inline):
(iso_allocate_primitive_zeroed_inline):
(iso_try_allocate_primitive_with_alignment_inline):
(iso_allocate_primitive_with_alignment_inline):
(iso_try_reallocate_primitive_inline):
(iso_reallocate_primitive_inline):
(iso_try_allocate_for_flex_inline):
* Source/bmalloc/libpas/src/libpas/iso_test_heap.c:
(iso_test_allocate_common_primitive):
(iso_test_allocate):
(iso_test_allocate_array_by_count):
* Source/bmalloc/libpas/src/libpas/iso_test_heap.h:
* Source/bmalloc/libpas/src/libpas/jit_heap.c:
(jit_heap_try_allocate):
* Source/bmalloc/libpas/src/libpas/jit_heap.h:
* Source/bmalloc/libpas/src/libpas/minalign32_heap.c:
(minalign32_allocate_common_primitive):
(minalign32_allocate):
(minalign32_allocate_array_by_count):
* Source/bmalloc/libpas/src/libpas/minalign32_heap.h:
* Source/bmalloc/libpas/src/libpas/pagesize64k_heap.c:
(pagesize64k_allocate_common_primitive):
(pagesize64k_allocate):
(pagesize64k_allocate_array_by_count):
* Source/bmalloc/libpas/src/libpas/pagesize64k_heap.h:
* Source/bmalloc/libpas/src/libpas/pas_allocation_mode.h:
* Source/bmalloc/libpas/src/libpas/thingy_heap.c:
(thingy_try_allocate_primitive):
(thingy_try_allocate_primitive_zeroed):
(thingy_try_allocate_primitive_with_alignment):
(thingy_try_reallocate_primitive):
(thingy_try_allocate):
(thingy_try_allocate_zeroed):
(thingy_try_allocate_array):
(thingy_try_allocate_zeroed_array):
(thingy_try_reallocate_array):
(thingy_utility_heap_allocate):
* Source/bmalloc/libpas/src/libpas/thingy_heap.h:
* Source/bmalloc/libpas/src/libpas/thingy_heap_prefix.h:
* Source/bmalloc/libpas/src/mbmalloc/mbmalloc_bmalloc.c:
(mbmalloc):
(mbmemalign):
(mbrealloc):
* Source/bmalloc/libpas/src/mbmalloc/mbmalloc_hotbit.c:
(mbmalloc):
(mbmemalign):
(mbrealloc):
* Source/bmalloc/libpas/src/mbmalloc/mbmalloc_iso_common_primitive.c:
(mbmalloc):
(mbmemalign):
(mbrealloc):
* Source/bmalloc/libpas/src/test/BitfitTests.cpp:
(std::testAllocateAlignedSmallerThanSizeClassAndSmallerThanLargestAvailable):
* Source/bmalloc/libpas/src/test/BmallocTests.cpp:
(std::testBmallocAllocate):
(std::testBmallocDeallocate):
* Source/bmalloc/libpas/src/test/EnumerationTests.cpp:
(std::testBasicEnumeration):
(std::testPGMEnumerationBasic):
(std::testPGMEnumerationAddAndFree):
* Source/bmalloc/libpas/src/test/ExpendableMemoryTests.cpp:
(std::testPayloadImpl):
(std::testRage):
(std::testRematerializeAfterSearchOfDecommitted):
(std::testBasicSizeClass):
* Source/bmalloc/libpas/src/test/HeapRefAllocatorIndexTests.cpp:
* Source/bmalloc/libpas/src/test/IsoDynamicPrimitiveHeapTests.cpp:
(std::allocate42):
(std::allocate42WithAlignment):
(std::allocate42Zeroed):
(std::reallocate42):
* Source/bmalloc/libpas/src/test/IsoHeapChaosTests.cpp:
(std::gigacageAllocate):
(std::testAllocationChaos):
* Source/bmalloc/libpas/src/test/IsoHeapPageSharingTests.cpp:
(std::testTakePages):
(std::testTakePagesFromCorrectHeap):
(std::testLargeHeapTakesPagesFromCorrectSmallHeap):
(std::testLargeHeapTakesPagesFromCorrectSmallHeapAllocateAfterFree):
(std::testLargeHeapTakesPagesFromCorrectSmallHeapWithFancyOrder):
(std::testLargeHeapTakesPagesFromCorrectLargeHeap):
(std::testLargeHeapTakesPagesFromCorrectLargeHeapAllocateAfterFreeOnSmallHeap):
(std::testLargeHeapTakesPagesFromCorrectLargeHeapAllocateAfterFreeOnAnotherLargeHeap):
(std::testLargeHeapTakesPagesFromCorrectLargeHeapWithFancyOrder):
(std::testSmallHeapTakesPagesFromCorrectLargeHeap):
(std::testSmallHeapTakesPagesFromCorrectLargeHeapWithFancyOrder):
(std::testSmallHeapTakesPagesFromCorrectLargeHeapAllocateAfterFreeOnSmallHeap):
(std::testSmallHeapTakesPagesFromCorrectLargeHeapAllocateAfterFreeOnAnotherLargeHeap):
(std::allocateThingiesImpl):
(std::testScavengerEventuallyReturnsMemory):
(std::testScavengerEventuallyReturnsMemoryEvenWithoutManualShrink):
* Source/bmalloc/libpas/src/test/IsoHeapPartialAndBaselineTests.cpp:
(std::testSimplePartialAllocations):
(std::testFreeAroundPrimordialStop):
(std::testFreeInterleavedAroundPrimordialStop):
(std::testMultiplePartialsFromDifferentHeapsPerShared):
(std::testMultiplePartialsFromDifferentThreadsPerShared):
(std::testTwoBaselinesEvictions):
* Source/bmalloc/libpas/src/test/IsoHeapReservedMemoryTests.cpp:
(std::testSizeProgression):
* Source/bmalloc/libpas/src/test/JITHeapTests.cpp:
(std::testAllocateShrinkAndAllocate):
(std::testAllocationSize):
* Source/bmalloc/libpas/src/test/LotsOfHeapsAndThreads.cpp:
(std::testLotsOfHeapsAndThreads):
* Source/bmalloc/libpas/src/test/MemalignTests.cpp:
* Source/bmalloc/libpas/src/test/PGMTests.cpp:
(std::testPGMSingleAlloc):
(std::testPGMMultipleAlloc):
(std::testPGMRealloc):
(std::testPGMErrors):
* Source/bmalloc/libpas/src/test/RaceTests.cpp:
(std::testLocalAllocatorStopRace):
(std::testLocalAllocatorStopRaceAgainstScavenge):
* Source/bmalloc/libpas/src/test/TLCDecommitTests.cpp:
(std::testTLCDecommit):
(std::testChaosThenDecommit):
(std::testAllocateFromStoppedBaselineImpl):
* Source/bmalloc/libpas/src/test/TSDTests.cpp:
(std::destructor):
(std::testTSD):
* Source/bmalloc/libpas/src/test/ThingyAndUtilityHeapAllocationTests.cpp:
(std::testFreeListRefillSpans):
(std::testInternalScavenge):
(std::testInternalScavengeFromCorrectDirectory):
(std::testSpuriousEligibility):
(std::testBasicSizeClassNotSet):
(std::testSmallDoubleFree):
(std::testSmallFreeInner):
(std::testSmallFreeNextWithShrink):
(std::testSmallFreeNextWithoutShrink):
(std::testSmallFreeNextBeforeShrink):
(std::testAllocationChaos):
(std::testCombinedAllocationChaos):
(std::testLargeDoubleFree):
(std::testLargeOffsetFree):
(std::testReallocatePrimitive):
(std::testReallocateArray):
* Source/bmalloc/libpas/src/test/ViewCacheTests.cpp:
(std::testDisableViewCacheUsingBoundForNoViewCache):
(std::testEnableViewCacheAtSomeBoundForNoViewCache):

Canonical link: https://commits.webkit.org/279280@main
